server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequestParameters.java

```diff
@@ -17,23 +17,26 @@
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toUnmodifiableMap;
+import static java.util.stream.Collectors.toUnmodifiableSet;
 
 /**
  * This class encapsulates the metrics and other information needed to define scope when we are requesting node stats.
  */
 public class NodesStatsRequestParameters implements Writeable {
     private CommonStatsFlags indices = new CommonStatsFlags();
-    private final Set<String> requestedMetrics = new HashSet<>();
+    private Set<Metric> requestedMetrics = new HashSet<>();
```

Comment:
nit: could we keep this `final`?

Reply:
Yes, I could.

I think in this case we do more work with keeping it final. When I parse transport message I already allocate a list of strings and a set of metrics. If  `requestedMetrics` is final I need to clear and copy from temporary set, while temporary set is what I need. It's still pretty small set, does not really matter if we copy elements. I dont have preference.

from below:
```
-        requestedMetrics.clear();
-        requestedMetrics.addAll(in.readStringCollectionAsList());
+        requestedMetrics = Metric.readSetFrom(in);
```

Reply:
That is a constructor. I think the same could be achieved with:
```
   private final Set<Metric> requestedMetrics;

    public NodesStatsRequestParameters() {
        requestedMetrics = new HashSet<>();
    }

    public NodesStatsRequestParameters(StreamInput in) throws IOException {
        indices = new CommonStatsFlags(in);
        requestedMetrics = Metric.readSetFrom(in);
...
```